src/tree/v2/azure/GroupingItem.ts

```diff
@@ -46,6 +48,14 @@ export class GroupingItem implements ResourceGroupsItem {
 
     async getChildren(): Promise<ResourceGroupsItem[] | undefined> {
         const sortedResources = this.resources.sort((a, b) => a.name.localeCompare(b.name));
+
+        this.onChangeBranchDataProviders((type: AzExtResourceType) => {
```

Comment:
Does it make more sense to do this on the root of the tree data provider? It could look for which branch data providers need to be refreshed once.

Like this, every branch data providers will have to do this map and refresh individually. I doubt the performance is really that bad, but it does feel a little wasteful.

Reply:
I initially tried to do it the way you describe. Basically expose a method like 
```typescript
getOnChangeBranchDataProviderForType(type: AzExtResourceType): Event<void>
```

which is called and listened to like

```typescript
getOnChangeBranchDataProviderForType(type)(() => {
	// do something
})
```

First of all, that second snippet is ugly IMO. And I couldn't find a nice way to avoid it. And more importantly, we'd have to keep a `Map<AzExtResourceType, Listener>`. Otherwise we would be just moving where the check happens, and not reducing the number of checks. I didn't think it was worth it considering this should have no real performance impact.

Reply:
Then would it make sense for the branch data providers to keep track what what types they own? This actually might be necessary. 

For example, if the Functions branch data provider doesn't have any resources yet and a refresh is fired, it wouldn't know to refresh itself because it wouldn't find any of that "type" in its `sortedResources`, right?

Reply:
Branch data providers aren't responsible for refreshing the list of resources. The list of resources is handled by the subscription items.

Reply:
Discussed offline.